src/ImageService.Broker/Consumers/CreateImagesMessageConsumer.cs

```diff
@@ -0,0 +1,80 @@
+﻿using LT.DigitalOffice.ImageService.Data.Interfaces;
+using LT.DigitalOffice.ImageService.Mappers.Db.Interfaces;
+using LT.DigitalOffice.ImageService.Mappers.Helpers.Interfaces;
+using LT.DigitalOffice.ImageService.Models.Db;
+using LT.DigitalOffice.Kernel.Broker;
+using LT.DigitalOffice.Models.Broker.Models;
+using LT.DigitalOffice.Models.Broker.Requests.Image;
+using LT.DigitalOffice.Models.Broker.Responses.Image;
+using MassTransit;
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace LT.DigitalOffice.ImageService.Broker.Consumers
+{
+    public class CreateImagesMessageConsumer : IConsumer<ICreateImagesMessageRequest>
+    {
+        private readonly IImageMessageRepository _imageMessageRepository;
+        private readonly IDbImageMessageMapper _mapper;
+        private readonly IResizeImageHelper _helper;
+
+        public CreateImagesMessageConsumer(
+            IImageMessageRepository imageMessageRepository,
+            IResizeImageHelper helper,
+            IDbImageMessageMapper mapper)
+        {
+            _imageMessageRepository = imageMessageRepository;
+            _helper = helper;
+            _mapper = mapper;
+        }
+
+        public async Task Consume(ConsumeContext<ICreateImagesMessageRequest> context)
+        {
+            object response = OperationResultWrapper.CreateResponse(CreateImages, context.Message);
+
+            await context.RespondAsync<IOperationResult<ICreateImagesResponse>>(response);
+        }
+
+        private object CreateImages(ICreateImagesMessageRequest request)
+        {
+            if (request.CreateImagesData == null)
+            {
+                return null;
+            }
+
+            List<DbImagesMessage> dbImages = new();
+            List<Guid> previewIds = new();
+            DbImagesMessage dbImageMessage;
+            DbImagesMessage dbPreviewImageMessage;
+            string resizedContent;
+
+            foreach (CreateImageData createImage in request.CreateImagesData)
+            {
+                dbImageMessage = _mapper.Map(createImage);
+                resizedContent = _helper.Resize(createImage.Content, createImage.Extension);
+
+                if (string.IsNullOrEmpty(resizedContent))
+                {
+                    dbImageMessage.ParentId = dbImageMessage.Id;
```

Comment:
Why? why not null?

Reply:
we decided to use null in parentId in full images
small images are previews and previews must have not null parentId